### PR TITLE
fusee-interface-tk: init at 1.0.0

### PR DESCRIPTION
**Pull request by Flameslice**
Add fusee-interfacee-tk package to applications/misc
Added self as maintainer to fusee-interface-tk

###### Motivation for this change
Wanted to use fusee-interface-tk from https://github.com/nh-server/fusee-interfacee-tk in NixOS

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
> Add fusee-interface-tk package to applications/misc
> Added self as maintainer to fusee-interface-tk

Adding yourself as a maintainer is preferred to be a separate commit.
Something like, `maintainers: add kristian-brucaj`. Then apply the fusee-interface-tk commit.

File: pkgs/applications/misc/fusee-interface-tk/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+let pythonEnv = python3.withPackages(ps: [ ps.tkinter ps.pyusb ]); 
+in stdenv.mkDerivation { 
+
+  pname = "fusee-interface-tk";
+  version = "1.0.0";
+  
+  src = fetchFromGitHub { 
+    owner = "nh-server";
+    repo = "fusee-interfacee-tk";
+    rev = "3053a5ffef8efb5ae3da9666f8f53285c5a904b2";
```
Comment (sikmir) [inline pkgs/applications/misc/fusee-interface-tk/default.nix]:
```suggestion
    rev = "V${version}";
```

File: pkgs/applications/misc/fusee-interface-tk/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+let pythonEnv = python3.withPackages(ps: [ ps.tkinter ps.pyusb ]); 
+in stdenv.mkDerivation { 
+
+  pname = "fusee-interface-tk";
+  version = "1.0.0";
+  
+  src = fetchFromGitHub { 
+    owner = "nh-server";
+    repo = "fusee-interfacee-tk";
```
Comment (sikmir) [inline pkgs/applications/misc/fusee-interface-tk/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/applications/misc/fusee-interface-tk/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    cp *.py $out/bin/ 
+    cp intermezzo.bin $out/bin/intermezzo.bin
+  '';
+
+  meta = with stdenv.lib; { 
+    homepage = "https://github.com/nh-server/fusee-interfacee-tk";
+    description = "A tool to send .bin files to a Nintendo Switch in RCM mode";
+    longDescription = "A mod of falquinhos Fusée Launcher for use with Nintendo Homebrew Switch Guide. It also adds the ability to mount SD while in RCM. 
+    Must be run as sudo.";
+    maintainers = [ applications.misc.kristian-brucaj ];
```
Comment (sikmir) [inline pkgs/applications/misc/fusee-interface-tk/default.nix]:
```suggestion
    maintainers = with maintainers; [ kristian-brucaj ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -3451,6 +3451,8 @@ in
   fuse-7z-ng = callPackage ../tools/filesystems/fuse-7z-ng { };
 
   fuse-overlayfs = callPackage ../tools/filesystems/fuse-overlayfs {};
+  
+  fusee-interface-tk = callPackage ../applications/misc/fusee-interface-tk { };
```
Comment (sikmir) [inline pkgs/top-level/all-packages.nix]: Consider using:
```suggestion
  fusee-interface-tk = python3Packages.callPackage ../applications/misc/fusee-interface-tk { };
```

File: pkgs/applications/misc/fusee-interface-tk/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{ stdenv , fetchFromGitHub , python3 , makeWrapper }: 
+
+let pythonEnv = python3.withPackages(ps: [ ps.tkinter ps.pyusb ]); 
+in stdenv.mkDerivation {
```
Comment (sikmir) [inline pkgs/applications/misc/fusee-interface-tk/default.nix]: Consider using:
```suggestion
buildPythonApplication rec { 
```

File: pkgs/applications/misc/fusee-interface-tk/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  
+  src = fetchFromGitHub { 
+    owner = "nh-server";
+    repo = "fusee-interfacee-tk";
+    rev = "3053a5ffef8efb5ae3da9666f8f53285c5a904b2";
+    sha256 = "0ycsxv71b5yvkcawxmcnmywxfvn8fdg1lyq71xdw7qrskxv5fgq7";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ pythonEnv ];
```
Comment (sikmir) [inline pkgs/applications/misc/fusee-interface-tk/default.nix]: Maybe something like:
```suggestion
  propagatedBuildInputs = [ tkinter pyusb ];
```

File: pkgs/applications/misc/fusee-interface-tk/default.nix (hunk 5 above)
Comment (Kristian-Brucaj) [reply pkgs/applications/misc/fusee-interface-tk/default.nix]: The reason I'm not using buildPythonApplication is that the package is structured oddly and lacks a setup.py. I plan to try and cobble one together later, but for now the stdenv.mkDerivation should be used. The same applies to the prior and next suggestions.

File: pkgs/applications/misc/fusee-interface-tk/default.nix (hunk 1 above)
Comment (Kristian-Brucaj) [reply pkgs/applications/misc/fusee-interface-tk/default.nix]: I can't seem to get this working on my end.

**Pull request closed without merging**